### ISSUE TITLE
Generalize the 'has input' conversion for 'binding' and all children

### ISSUE THREAD

**Issue by vanaukenk** [geneontology/gocamgen#50]
Doing some QC on the imports, I realized that WB (and others) have annotations to other children of 'binding' (GO:0005488) that use the IPI (ECO:0000353) evidence code and contain an entry in the With/From field.

To be consistent in how we model 'binding' annotations in GO-CAM, we'll need to broaden our conversion rule so that any annotation to GO:0005488 that has an entry in the With/From field, will convert the With/From value to a 'has input' annotation extension, just like we're doing for 'protein binding' (GO:0005515) and its children.

For an example, see the 'miRNA binding' (GO:0035198) annotation in this [alg-1 model](http://noctua-dev.berkeleybop.org/editor/graph/gomodel:2012529b-024e-44ab-8c20-dcc135849589).

@dustine32 - let me know if you need any more info or examples.  Thx.

**Comment by dustine32** [comment]
@vanaukenk Good call! I've occasionally been seeing some DNA binding examples and was wondering if they should get the same treatment.

Should be a one-line change. Maybe even make this "top-level binding term" MOD-specific configurable like the filter rules?

**Comment by vanaukenk** [comment]
@dustine32 - I think we can go ahead and make this a general rule across all the MODs.  

If a curator annotated to 'binding' or a child and entered something in the With/From field, we will want to translate all of those value to 'has input'.

@ukemi - sound okay to you?

**Comment by dustine32** [comment]
Changed root term (used for determining whether to translate with/from to `has_input` edges) to "binding" GO:0005488 whereas it was previously "protein binding" GO:0005515. You can see this now works for other terms like "DNA binding":
![image](https://user-images.githubusercontent.com/2678599/58595257-ec83b980-8224-11e9-99ba-d10db406e0f9.png)
In the model for [MGI:MGI:97747](http://68.181.46.18:8910/editor/graph/gomodel:dbaff01e-a8c5-4074-a539-288ddab207c3) from line:
```
MGI	MGI:97747	enables	GO:0003677	MGI:MGI:3529655|PMID:15681609	ECO:0000353	UniProtKB:P28700		20050715	MGI
```

**Comment by vanaukenk** [comment]
This looks okay to me on the USC server.  I'd like to double-check other models, but they should be fine.  Thanks!

**Comment by ukemi** [comment]
The translation looks good. The annotation looks strange. I may alter this one.

**Comment by vanaukenk** [comment]
Note that we still need to work with @balhoff to make sure the GPAD output works as we want.

**Comment by dustine32** [comment]
Finally found an example of a DNA binding IPI in noctua-dev: [MGI:MGI:1096381](http://noctua-dev.berkeleybop.org/editor/graph/gomodel:71365089-2cab-4707-9895-5f6d526bccba)
![image](https://user-images.githubusercontent.com/2678599/68516033-c5033500-0237-11ea-9a16-05efebb6fd23.png)
From line:
```
MGI MGI:1096381 enables GO:0003677 MGI:MGI:1276614|PMID:9704006 ECO:0000353 UniProtKB:Q61221 20110329 MGI
```
@vanaukenk @ukemi If this looks good we can finally close this ticket!

**Comment by ukemi** [comment]
I think that there is an issue with this that I pointed out in the example in #37. We should only use this conversion for binding annotations that use the IPI evidence code. If they use another evidence code, like ISO, they should be retained in the 'with' field.
